packages/graph/hash_graph/Makefile.toml

```diff
@@ -24,11 +24,21 @@ dependencies = ["install-cargo-nextest"]
 [tasks.test-rest-api]
 # This is a temporary solution until we have e2e tests in place
 script = """
+#!/usr/bin/env bash
+
+# Fail entire script if any error occurs.
+set -eo pipefail
+
 cargo build
 cargo run&
 pid=$!
+# ensure pid is killed no matter what happens
+trap "kill $pid" 0 ERR
+
+exit_status=0
+
 yarn httpyac send --all ${CARGO_MAKE_CURRENT_TASK_INITIAL_MAKEFILE_DIRECTORY}/tests/rest-test.http
-ret=$?
-kill "$pid"
-exit $ret
+
+# This generates a `openapitools.json` file.
```

Comment:
I don't understand the comment really, is it just warning there's a side-effect? Because when reading it I thought it was saying the purpose of the line

Reply:
Yeah it's a side-effect. Let me rephrase it. I tried getting around the issue by adding the generated file to gitignore

Reply:
Addressed in b9e95694f45624b0950bdb78e11ceb1a6c6fd271

Reply:
We could just `rm` it but I don't feel too strongly about this

Reply:
Let me just have a look. There's a couple annoyances with this file, one being that it is created at the root and not in hash_graph.

Reply:
Addressed in 5fc1529c0ccfa5b5c030300147c9386ba3db1314